core/src/main/java/hudson/tools/ToolInstallation.java

```diff
@@ -210,13 +221,32 @@ protected String translateFor(Node node, TaskListener log) throws IOException, I
      * Invoked by XStream when this object is read into memory.
      */
     protected Object readResolve() {
-        if(properties==null)
-            properties = new DescribableList<>(Saveable.NOOP);
-        for (ToolProperty<?> p : properties)
-            _setTool(p, this);
+        if (properties != null) {
+            for (ToolProperty<?> p : properties) {
+                _setTool(p, this);
+            }
+        }
         return this;
     }
 
+    protected Object writeReplace() throws Exception {
+        if (Channel.current() == null) { // XStream
```

Comment:
Nice!

Reply:
Well, not nice, but seems to work well enough. I am trying to fix popular plugins to not need this hack.